### ISSUE TITLE
jQA 2.0 does not run with Java SE 17

### ISSUE THREAD

**Issue by JanWesterkamp-iJUG** [jqassistant-tutorials/jakarta-ee-dependencies#9]
When run with Java SE 17 (as before with version 1.2 successfully, meaning the config arguments are added), there is an issue to load the nio module somehow. Log:

...
[[1;34mINFO[m] Connecting to store at 'file:/Users/jan/git/jakarta-ee-dependencies/target/jqassistant/store/'
[[1;36mDEBUG[m] try to lookup provider-class com.buschmais.xo.neo4j.embedded.api.FileDatabaseManagementServiceBuilderFactory
[[1;36mDEBUG[m] Creating graph database service datastore for directory '/Users/jan/git/jakarta-ee-dependencies/target/jqassistant/store'.
[[1;34mINFO[m] [1m------------------------------------------------------------------------[m
[[1;34mINFO[m] [1;31mBUILD FAILURE[m
[[1;34mINFO[m] [1m------------------------------------------------------------------------[m
[[1;34mINFO[m] Total time:  3.169 s
[[1;34mINFO[m] Finished at: 2023-05-16T21:10:03+02:00
[[1;34mINFO[m] [1m------------------------------------------------------------------------[m
[[1;31mERROR[m] Failed to execute goal [32mcom.buschmais.jqassistant:jqassistant-maven-plugin:2.0.1:scan[m [1m(default-cli)[m on project [36mjakarta-ee-dependencies[m: [1;31mExecution default-cli of goal com.buschmais.jqassistant:jqassistant-maven-plugin:2.0.1:scan failed: An API incompatibility was encountered while executing com.buschmais.jqassistant:jqassistant-maven-plugin:2.0.1:scan: java.lang.LinkageError: Cannot to link java.nio.DirectByteBuffer[m
[[1;31mERROR[m] [1;31m-----------------------------------------------------[m
[[1;31mERROR[m] [1;31mrealm =    plugin>com.buschmais.jqassistant:jqassistant-maven-plugin:2.0.1[m
[[1;31mERROR[m] [1;31mstrategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy[m
[[1;31mERROR[m] [1;31murls[0] = file:/Users/jan/.m2/repository/com/buschmais/jqassistant/jqassistant-maven-plugin/2.0.1/jqassistant-maven-plugin-2.0.1.jar[m

...

[[1;31mERROR[m] [1;31murls[262] = file:/Users/jan/.m2/repository/org/apache/commons/commons-lang3/3.12.0/commons-lang3-3.12.0.jar[m
[[1;31mERROR[m] [1;31mNumber of foreign imports: 1[m
[[1;31mERROR[m] [1;31mimport: Entry[import  from realm ClassRealm[maven.api, parent: null]][m
[[1;31mERROR[m] [1;31m[m
[[1;31mERROR[m] [1;31m-----------------------------------------------------[m
[[1;31mERROR[m] [1;31m[m: module java.base does not open java.nio to unnamed module @3313d477
[[1;31mERROR[m] -> [1m[Help 1][m
[1;31morg.apache.maven.lifecycle.LifecycleExecutionException[m: [1;31mFailed to execute goal [32mcom.buschmais.jqassistant:jqassistant-maven-plugin:2.0.1:scan[m [1m(default-cli)[m on project [36mjakarta-ee-dependencies[m: [1;31mExecution default-cli of goal com.buschmais.jqassistant:jqassistant-maven-plugin:2.0.1:scan failed: An API incompatibility was encountered while executing com.buschmais.jqassistant:jqassistant-maven-plugin:2.0.1:scan: java.lang.LinkageError: Cannot to link java.nio.DirectByteBuffer
-----------------------------------------------------
realm =    plugin>com.buschmais.jqassistant:jqassistant-maven-plugin:2.0.1
strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
urls[0] = file:/Users/jan/.m2/repository/com/buschmais/jqassistant/jqassistant-maven-plugin/2.0.1/jqassistant-maven-plugin-2.0.1.jar

...

urls[262] = file:/Users/jan/.m2/repository/org/apache/commons/commons-lang3/3.12.0/commons-lang3-3.12.0.jar
Number of foreign imports: 1
import: Entry[import  from realm ClassRealm[maven.api, parent: null]]

-----------------------------------------------------
[m[m
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 ([1mMojoExecutor.java:375[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.doExecute ([1mMojoExecutor.java:351[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:215[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:171[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:163[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject ([1mLifecycleModuleBuilder.java:117[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject ([1mLifecycleModuleBuilder.java:81[m)
    [1mat[m org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build ([1mSingleThreadedBuilder.java:56[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleStarter.execute ([1mLifecycleStarter.java:128[m)
    [1mat[m org.apache.maven.DefaultMaven.doExecute ([1mDefaultMaven.java:298[m)
    [1mat[m org.apache.maven.DefaultMaven.doExecute ([1mDefaultMaven.java:192[m)
    [1mat[m org.apache.maven.DefaultMaven.execute ([1mDefaultMaven.java:105[m)
    [1mat[m org.apache.maven.cli.MavenCli.execute ([1mMavenCli.java:960[m)
    [1mat[m org.apache.maven.cli.MavenCli.doMain ([1mMavenCli.java:293[m)
    [1mat[m org.apache.maven.cli.MavenCli.main ([1mMavenCli.java:196[m)
    [1mat[m jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 ([1mNative Method[m)
    [1mat[m jdk.internal.reflect.NativeMethodAccessorImpl.invoke ([1mNativeMethodAccessorImpl.java:77[m)
    [1mat[m jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke ([1mDelegatingMethodAccessorImpl.java:43[m)
    [1mat[m java.lang.reflect.Method.invoke ([1mMethod.java:568[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced ([1mLauncher.java:282[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.launch ([1mLauncher.java:225[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode ([1mLauncher.java:406[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.main ([1mLauncher.java:347[m)
[1mCaused by[m: org.apache.maven.plugin.PluginExecutionException: [1;31mExecution default-cli of goal com.buschmais.jqassistant:jqassistant-maven-plugin:2.0.1:scan failed: An API incompatibility was encountered while executing com.buschmais.jqassistant:jqassistant-maven-plugin:2.0.1:scan: java.lang.LinkageError: Cannot to link java.nio.DirectByteBuffer
-----------------------------------------------------
realm =    plugin>com.buschmais.jqassistant:jqassistant-maven-plugin:2.0.1
strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
urls[0] = file:/Users/jan/.m2/repository/com/buschmais/jqassistant/jqassistant-maven-plugin/2.0.1/jqassistant-maven-plugin-2.0.1.jar

...

urls[262] = file:/Users/jan/.m2/repository/org/apache/commons/commons-lang3/3.12.0/commons-lang3-3.12.0.jar
Number of foreign imports: 1
import: Entry[import  from realm ClassRealm[maven.api, parent: null]]

-----------------------------------------------------
[m
    [1mat[m org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo ([1mDefaultBuildPluginManager.java:186[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 ([1mMojoExecutor.java:370[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.doExecute ([1mMojoExecutor.java:351[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:215[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:171[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:163[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject ([1mLifecycleModuleBuilder.java:117[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject ([1mLifecycleModuleBuilder.java:81[m)
    [1mat[m org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build ([1mSingleThreadedBuilder.java:56[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleStarter.execute ([1mLifecycleStarter.java:128[m)
    [1mat[m org.apache.maven.DefaultMaven.doExecute ([1mDefaultMaven.java:298[m)
    [1mat[m org.apache.maven.DefaultMaven.doExecute ([1mDefaultMaven.java:192[m)
    [1mat[m org.apache.maven.DefaultMaven.execute ([1mDefaultMaven.java:105[m)
    [1mat[m org.apache.maven.cli.MavenCli.execute ([1mMavenCli.java:960[m)
    [1mat[m org.apache.maven.cli.MavenCli.doMain ([1mMavenCli.java:293[m)
    [1mat[m org.apache.maven.cli.MavenCli.main ([1mMavenCli.java:196[m)
    [1mat[m jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 ([1mNative Method[m)
    [1mat[m jdk.internal.reflect.NativeMethodAccessorImpl.invoke ([1mNativeMethodAccessorImpl.java:77[m)
    [1mat[m jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke ([1mDelegatingMethodAccessorImpl.java:43[m)
    [1mat[m java.lang.reflect.Method.invoke ([1mMethod.java:568[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced ([1mLauncher.java:282[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.launch ([1mLauncher.java:225[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode ([1mLauncher.java:406[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.main ([1mLauncher.java:347[m)
[1mCaused by[m: org.apache.maven.plugin.PluginContainerException: [1;31mAn API incompatibility was encountered while executing com.buschmais.jqassistant:jqassistant-maven-plugin:2.0.1:scan: java.lang.LinkageError: Cannot to link java.nio.DirectByteBuffer
-----------------------------------------------------
realm =    plugin>com.buschmais.jqassistant:jqassistant-maven-plugin:2.0.1
strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
urls[0] = file:/Users/jan/.m2/repository/com/buschmais/jqassistant/jqassistant-maven-plugin/2.0.1/jqassistant-maven-plugin-2.0.1.jar

...

urls[262] = file:/Users/jan/.m2/repository/org/apache/commons/commons-lang3/3.12.0/commons-lang3-3.12.0.jar
Number of foreign imports: 1
import: Entry[import  from realm ClassRealm[maven.api, parent: null]]

-----------------------------------------------------
[m
    [1mat[m org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo ([1mDefaultBuildPluginManager.java:184[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 ([1mMojoExecutor.java:370[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.doExecute ([1mMojoExecutor.java:351[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:215[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:171[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:163[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject ([1mLifecycleModuleBuilder.java:117[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject ([1mLifecycleModuleBuilder.java:81[m)
    [1mat[m org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build ([1mSingleThreadedBuilder.java:56[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleStarter.execute ([1mLifecycleStarter.java:128[m)
    [1mat[m org.apache.maven.DefaultMaven.doExecute ([1mDefaultMaven.java:298[m)
    [1mat[m org.apache.maven.DefaultMaven.doExecute ([1mDefaultMaven.java:192[m)
    [1mat[m org.apache.maven.DefaultMaven.execute ([1mDefaultMaven.java:105[m)
    [1mat[m org.apache.maven.cli.MavenCli.execute ([1mMavenCli.java:960[m)
    [1mat[m org.apache.maven.cli.MavenCli.doMain ([1mMavenCli.java:293[m)
    [1mat[m org.apache.maven.cli.MavenCli.main ([1mMavenCli.java:196[m)
    [1mat[m jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 ([1mNative Method[m)
    [1mat[m jdk.internal.reflect.NativeMethodAccessorImpl.invoke ([1mNativeMethodAccessorImpl.java:77[m)
    [1mat[m jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke ([1mDelegatingMethodAccessorImpl.java:43[m)
    [1mat[m java.lang.reflect.Method.invoke ([1mMethod.java:568[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced ([1mLauncher.java:282[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.launch ([1mLauncher.java:225[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode ([1mLauncher.java:406[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.main ([1mLauncher.java:347[m)
[1mCaused by[m: java.lang.LinkageError: [1;31mCannot to link java.nio.DirectByteBuffer[m
    [1mat[m org.neo4j.internal.unsafe.UnsafeUtil.<clinit> ([1mUnsafeUtil.java:128[m)
    [1mat[m org.neo4j.memory.RuntimeInternals.guessHeaderSize ([1mRuntimeInternals.java:158[m)
    [1mat[m org.neo4j.memory.RuntimeInternals.<clinit> ([1mRuntimeInternals.java:53[m)
    [1mat[m org.neo4j.memory.HeapEstimator.<clinit> ([1mHeapEstimator.java:103[m)
    [1mat[m org.neo4j.internal.collector.RecentQueryBuffer.<clinit> ([1mRecentQueryBuffer.java:37[m)
    [1mat[m org.neo4j.graphdb.factory.module.GlobalModule.<init> ([1mGlobalModule.java:212[m)
    [1mat[m org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.createGlobalModule ([1mDatabaseManagementServiceFactory.java:272[m)
    [1mat[m org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.build ([1mDatabaseManagementServiceFactory.java:133[m)
    [1mat[m org.neo4j.dbms.api.DatabaseManagementServiceBuilder.newDatabaseManagementService ([1mDatabaseManagementServiceBuilder.java:101[m)
    [1mat[m org.neo4j.dbms.api.DatabaseManagementServiceBuilder.build ([1mDatabaseManagementServiceBuilder.java:94[m)
    [1mat[m com.buschmais.xo.neo4j.embedded.api.EmbeddedNeo4jXOProvider.createDatastore ([1mEmbeddedNeo4jXOProvider.java:39[m)
    [1mat[m com.buschmais.xo.impl.XOManagerFactoryImpl.<init> ([1mXOManagerFactoryImpl.java:48[m)
    [1mat[m com.buschmais.xo.impl.bootstrap.XOBootstrapServiceImpl.createXOManagerFactory ([1mXOBootstrapServiceImpl.java:41[m)
    [1mat[m com.buschmais.xo.api.bootstrap.XO.createXOManagerFactory ([1mXO.java:48[m)
    [1mat[m com.buschmais.jqassistant.core.store.impl.AbstractGraphStore.start ([1mAbstractGraphStore.java:66[m)
    [1mat[m com.buschmais.jqassistant.scm.maven.provider.CachingStoreProvider.getStore ([1mCachingStoreProvider.java:74[m)
    [1mat[m com.buschmais.jqassistant.scm.maven.AbstractMojo.getStore ([1mAbstractMojo.java:212[m)
    [1mat[m com.buschmais.jqassistant.scm.maven.AbstractMojo.withStore ([1mAbstractMojo.java:172[m)
    [1mat[m com.buschmais.jqassistant.scm.maven.AbstractMojo.execute ([1mAbstractMojo.java:120[m)
    [1mat[m org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo ([1mDefaultBuildPluginManager.java:137[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 ([1mMojoExecutor.java:370[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.doExecute ([1mMojoExecutor.java:351[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:215[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:171[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:163[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject ([1mLifecycleModuleBuilder.java:117[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject ([1mLifecycleModuleBuilder.java:81[m)
    [1mat[m org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build ([1mSingleThreadedBuilder.java:56[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleStarter.execute ([1mLifecycleStarter.java:128[m)
    [1mat[m org.apache.maven.DefaultMaven.doExecute ([1mDefaultMaven.java:298[m)
    [1mat[m org.apache.maven.DefaultMaven.doExecute ([1mDefaultMaven.java:192[m)
    [1mat[m org.apache.maven.DefaultMaven.execute ([1mDefaultMaven.java:105[m)
    [1mat[m org.apache.maven.cli.MavenCli.execute ([1mMavenCli.java:960[m)
    [1mat[m org.apache.maven.cli.MavenCli.doMain ([1mMavenCli.java:293[m)
    [1mat[m org.apache.maven.cli.MavenCli.main ([1mMavenCli.java:196[m)
    [1mat[m jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 ([1mNative Method[m)
    [1mat[m jdk.internal.reflect.NativeMethodAccessorImpl.invoke ([1mNativeMethodAccessorImpl.java:77[m)
    [1mat[m jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke ([1mDelegatingMethodAccessorImpl.java:43[m)
    [1mat[m java.lang.reflect.Method.invoke ([1mMethod.java:568[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced ([1mLauncher.java:282[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.launch ([1mLauncher.java:225[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode ([1mLauncher.java:406[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.main ([1mLauncher.java:347[m)
[1mCaused by[m: java.lang.IllegalAccessException: [1;31mmodule java.base does not open java.nio to unnamed module @3313d477[m
    [1mat[m java.lang.invoke.MethodHandles.privateLookupIn ([1mMethodHandles.java:259[m)
    [1mat[m org.neo4j.internal.unsafe.UnsafeUtil.<clinit> ([1mUnsafeUtil.java:111[m)
    [1mat[m org.neo4j.memory.RuntimeInternals.guessHeaderSize ([1mRuntimeInternals.java:158[m)
    [1mat[m org.neo4j.memory.RuntimeInternals.<clinit> ([1mRuntimeInternals.java:53[m)
    [1mat[m org.neo4j.memory.HeapEstimator.<clinit> ([1mHeapEstimator.java:103[m)
    [1mat[m org.neo4j.internal.collector.RecentQueryBuffer.<clinit> ([1mRecentQueryBuffer.java:37[m)
    [1mat[m org.neo4j.graphdb.factory.module.GlobalModule.<init> ([1mGlobalModule.java:212[m)
    [1mat[m org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.createGlobalModule ([1mDatabaseManagementServiceFactory.java:272[m)
    [1mat[m org.neo4j.graphdb.facade.DatabaseManagementServiceFactory.build ([1mDatabaseManagementServiceFactory.java:133[m)
    [1mat[m org.neo4j.dbms.api.DatabaseManagementServiceBuilder.newDatabaseManagementService ([1mDatabaseManagementServiceBuilder.java:101[m)
    [1mat[m org.neo4j.dbms.api.DatabaseManagementServiceBuilder.build ([1mDatabaseManagementServiceBuilder.java:94[m)
    [1mat[m com.buschmais.xo.neo4j.embedded.api.EmbeddedNeo4jXOProvider.createDatastore ([1mEmbeddedNeo4jXOProvider.java:39[m)
    [1mat[m com.buschmais.xo.impl.XOManagerFactoryImpl.<init> ([1mXOManagerFactoryImpl.java:48[m)
    [1mat[m com.buschmais.xo.impl.bootstrap.XOBootstrapServiceImpl.createXOManagerFactory ([1mXOBootstrapServiceImpl.java:41[m)
    [1mat[m com.buschmais.xo.api.bootstrap.XO.createXOManagerFactory ([1mXO.java:48[m)
    [1mat[m com.buschmais.jqassistant.core.store.impl.AbstractGraphStore.start ([1mAbstractGraphStore.java:66[m)
    [1mat[m com.buschmais.jqassistant.scm.maven.provider.CachingStoreProvider.getStore ([1mCachingStoreProvider.java:74[m)
    [1mat[m com.buschmais.jqassistant.scm.maven.AbstractMojo.getStore ([1mAbstractMojo.java:212[m)
    [1mat[m com.buschmais.jqassistant.scm.maven.AbstractMojo.withStore ([1mAbstractMojo.java:172[m)
    [1mat[m com.buschmais.jqassistant.scm.maven.AbstractMojo.execute ([1mAbstractMojo.java:120[m)
    [1mat[m org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo ([1mDefaultBuildPluginManager.java:137[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 ([1mMojoExecutor.java:370[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.doExecute ([1mMojoExecutor.java:351[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:215[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:171[m)
    [1mat[m org.apache.maven.lifecycle.internal.MojoExecutor.execute ([1mMojoExecutor.java:163[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject ([1mLifecycleModuleBuilder.java:117[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject ([1mLifecycleModuleBuilder.java:81[m)
    [1mat[m org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build ([1mSingleThreadedBuilder.java:56[m)
    [1mat[m org.apache.maven.lifecycle.internal.LifecycleStarter.execute ([1mLifecycleStarter.java:128[m)
    [1mat[m org.apache.maven.DefaultMaven.doExecute ([1mDefaultMaven.java:298[m)
    [1mat[m org.apache.maven.DefaultMaven.doExecute ([1mDefaultMaven.java:192[m)
    [1mat[m org.apache.maven.DefaultMaven.execute ([1mDefaultMaven.java:105[m)
    [1mat[m org.apache.maven.cli.MavenCli.execute ([1mMavenCli.java:960[m)
    [1mat[m org.apache.maven.cli.MavenCli.doMain ([1mMavenCli.java:293[m)
    [1mat[m org.apache.maven.cli.MavenCli.main ([1mMavenCli.java:196[m)
    [1mat[m jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 ([1mNative Method[m)
    [1mat[m jdk.internal.reflect.NativeMethodAccessorImpl.invoke ([1mNativeMethodAccessorImpl.java:77[m)
    [1mat[m jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke ([1mDelegatingMethodAccessorImpl.java:43[m)
    [1mat[m java.lang.reflect.Method.invoke ([1mMethod.java:568[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced ([1mLauncher.java:282[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.launch ([1mLauncher.java:225[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode ([1mLauncher.java:406[m)
    [1mat[m org.codehaus.plexus.classworlds.launcher.Launcher.main ([1mLauncher.java:347[m)
[[1;31mERROR[m] 
[[1;31mERROR[m] 
[[1;31mERROR[m] For more information about the errors and possible solutions, please read the following articles:
[[1;31mERROR[m] [1m[Help 1][m http://cwiki.apache.org/confluence/display/MAVEN/PluginContainerException

Looks like the NIO module has some issue:

module java.base does not open java.nio to unnamed module @3313d477

And that causes this issue:

Cannot to link java.nio.DirectByteBuffer

Changing to Java SE 11 shows some related warnings:

...
[[1;34mINFO[m] Connecting to store at 'file:/Users/jan/git/jakarta-ee-dependencies/target/jqassistant/store/'
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access using Lookup on org.neo4j.internal.unsafe.UnsafeUtil (file:/Users/jan/.m2/repository/org/neo4j/neo4j-unsafe/4.4.20/neo4j-unsafe-4.4.20.jar) to class java.nio.Buffer
WARNING: Please consider reporting this to the maintainers of org.neo4j.internal.unsafe.UnsafeUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[[1;34mINFO[m] Initializing embedded Neo4j server 4.x
[[1;34mINFO[m] Resetting store.
...

**Comment by DirkMahler** [comment]
I added a JVM configuration for Maven, should work now. Can you verify this?

**Comment by DirkMahler** [comment]
Moving the plugin declarations to a separate configuration has been done for two reasons:

- Declaring the plugins as dependencies to the jQA Maven plugin required to have a consistent setup through all modules of a maven reactor, i.e. by having a consistent hierarchy. Sadly there are quite some reactors who do not provide this precondition. Having the configuration separatly allows using jQA with any reactor without modifying a pom.xml by just running the plugin goals directly from the command line
- As a positive side effect the configurations for the jQA Maven plugin and the jQA CLI are now nearly identical

Initially the idea was to keep the configuration section in the pom.xml in parallel to the YAML file (just another Microprofile config source) but it turned out that the YAML file is much easier and more consistent to use, so we decided to remove the pom.xml based configuration completely.  Hence that properties specified in Maven are still evaluated (and furthermoresystem properties and environment variables).

**Comment by DirkMahler** [comment]
BTW: Under the hood we're now using SmallRye Config (https://smallrye.io/smallrye-config/Main/), an implementation of Eclipse Microprofile Config - and it does a real good job!

**Comment by JanWesterkamp-iJUG** [comment]
@DirkMahler : That's nice, that will make the MicroProfilers happy! :-)

But the analysis of MP is broken somehow - I created a separate issue #11 for this.

I also would like to update the configurations and add some separate specs and their TCK to determin the (real) release order (waves) for Jakarta. In some cases the API has or or less dependencies compared to the TCK...
First step could be analysing the TCK artifact too to update release order manually.
Of course this is only possible for specs that have a separated TCK already (this is ongoing work for all the specs at the moment).

**Comment by DirkMahler** [comment]
Closing this issue as the remaining problem is covered by issues #11 .
